core/trino-parser/src/main/java/io/trino/sql/tree/QualifiedName.java

```diff
@@ -30,8 +29,11 @@
 
 public class QualifiedName
 {
-    private final List<String> parts;
     private final List<Identifier> originalParts;
+    private final List<String> parts;
+    private final Optional<QualifiedName> prefix;
+    private final String suffix;
+    private final String name;
```

Comment:
- Document why these are not part of equals, hashCode
- Document why these are precomputed in constructor. Do we have benchmark results supporting this change?